lib/plugins/cpuUsageThrottle.js

```diff
@@ -0,0 +1,235 @@
+var defaultResponse = errors.ServiceUnavailableError;
+var EWMA = require('ewma');
+
+/**
+ * cpuUsageThrottle
+ *
+ * cpuUsageThrottle is a middleware that rejects a variable number of requests
+ * (between 0% and 100%) based on a historical view of CPU utilization of a
+ * Node.js process. Essentially, this plugin allows you to define what
+ * constitutes a saturated Node.js process via CPU utilization and it will
+ * handle dropping a % of requests based on that definiton. This is useful when
+ * you would like to keep CPU bound tasks from piling up causing an increased
+ * per-request latency.
+ *
+ * The algorithm asks you for a maximum CPU utilization rate, which it uses to
+ * determine at what point it should be rejecting 100% of traffic. For a normal
+ * Node.js service, this is 1 since Node is single threaded. It uses this,
+ * paired with a limit that you provide to determine the total % of traffic it
+ * should be rejecting. For example, if you specify a limit of .5 and a max of
+ * 1, and the current EWMA (next paragraph) value reads .75, this plugin will
+ * reject approximately 50% of all requests.
+ *
+ * When looking at the process' CPU usage, this algorithm will take a load
+ * average over a user specified interval. This is similar to Linux's top. For
+ * example, if given an interval of 250ms, this plugin will attempt to record
+ * the average CPU utilization over 250ms intervals. Due to contention for
+ * resources, the duration of each average may be wider or narrower than 250ms.
+ * To compensate for this, we use an exponentially weighted moving average.
+ * The EWMA algorithm is provided by the ewma module. The parameter for
+ * configuring the EWMA is halfLife. This value controls how quickly each
+ * load average measurment decays to half it's value when being represented in
+ * the current average. For example, if you have an interval of 250, and a
+ * halfLife of 250, you will take the previous ewma value multiplied by  .5 and
+ * add it to the new CPU utilization average measurement multiplied by .5. The
+ * previous value and the new measurement would each represent 50% of the new
+ * value. A good way of thinking about the halfLife is in terms of how
+ * responsive this plugin will be to spikes in CPU utilization. The higher the
+ * halfLife, the longer CPU utilization will have to remain above your defined
+ * limit before this plugin begins rejecting requests and, converserly, the
+ * longer it will have to drop below your limit before the plugin begins
+ * accepting requests again. This is a knob you will want to with play when
+ * trying to determine the ideal value for your use case.
+ *
+ * For a better understanding of the EWMA algorithn, refer to the documentation
+ * for the ewma module.
+ *
+ * @param {Object} opts Configure this plugin.
+ * @param {Number} opts.limit The point at which restify will begin rejecting a
+ *    % of all requests at the front door. This value is a percentage like you
```

Comment:
I think here it would be less confusing and more precise to just use the temr `CPU utilization`, and link to documentation: http://www.brendangregg.com/blog/2017-05-09/cpu-utilization-is-wrong.html

Reply:
Agreed this is confusing. I removed the reference to linux top for the reasons detailed in https://github.com/restify/node-restify/pull/1460#discussion_r137885294